api/src/main/java/org/pmiops/workbench/db/dao/DataSetDao.java

```diff
@@ -16,6 +17,20 @@
   List<DbDataset> findByWorkspaceId(long workspaceId);
 
   default Map<Boolean, Long> getInvalidToCountMap() {
-    return DaoUtils.getAttributeToCountMap(findAll(), DbDataset::getInvalid);
+    final List<InvalidToCountResult> rows = getInvalidToCount();
+    return rows.stream()
+        .collect(
+            ImmutableMap.toImmutableMap(
+                InvalidToCountResult::getIsInvalid, InvalidToCountResult::getInvalidCount));
+  }
+
+  @Query(
+      "SELECT invalid, count(dataSetId) AS invalidCount FROM DbDataset GROUP BY invalid ORDER BY invalid")
+  List<InvalidToCountResult> getInvalidToCount();
```

Comment:
Can we write a test class for this in `DataSetDaoTest` and test all the methods?

Reply:
For scope reasons,  I'm going to just. cover the methods added here for now.